src/createRouter.ts

```diff
@@ -121,6 +143,21 @@ const createRouter: RouterFactory = compose<RouterMixin, RouterOptions>({
 		}
 	},
 
+	observeHistory(history: History, context: Context, dispatchInitial: boolean): PausableHandle {
+		if (!historyMap.has(this)) {
+			const listener = pausable(history, 'change', (event: HistoryChangeEvent) => {
+				this.dispatch(context, event.value);
+			});
+			historyMap.set(this, { history, listener, context });
+			if (dispatchInitial) {
+				this.dispatch(context, history.current);
+			}
+			this.own(listener);
+			this.own(history);
+			return listener;
```

Comment:
Maybe… though if `history` or `context` are different that may warrant an exception. So maybe just throw if the router is already observing history?